extensions/control-plane/edr/edr-store-receiver/src/main/java/org/eclipse/edc/connector/edr/store/receiver/EndpointDataReferenceStoreReceiver.java

```diff
@@ -0,0 +1,135 @@
+ *
+ */
+
+package org.eclipse.edc.connector.edr.store.receiver;
+
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
+import org.eclipse.edc.connector.policy.spi.store.PolicyArchive;
+import org.eclipse.edc.connector.spi.contractagreement.ContractAgreementService;
+import org.eclipse.edc.connector.transfer.spi.event.TransferProcessCompleted;
+import org.eclipse.edc.connector.transfer.spi.event.TransferProcessEvent;
+import org.eclipse.edc.connector.transfer.spi.event.TransferProcessStarted;
+import org.eclipse.edc.connector.transfer.spi.event.TransferProcessTerminated;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcess;
+import org.eclipse.edc.edr.spi.store.EndpointDataReferenceStore;
+import org.eclipse.edc.edr.spi.types.EndpointDataReferenceEntry;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.event.Event;
+import org.eclipse.edc.spi.event.EventEnvelope;
+import org.eclipse.edc.spi.event.EventSubscriber;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.transaction.spi.TransactionContext;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class EndpointDataReferenceStoreReceiver implements EventSubscriber {
+
+    private final EndpointDataReferenceStore dataReferenceStore;
+    private final PolicyArchive policyArchive;
+    private final ContractAgreementService contractAgreementService;
+    private final TransactionContext transactionContext;
+    private final Monitor monitor;
+    private final Map<Class<? extends TransferProcessEvent>, Function<? extends TransferProcessEvent, Result<Void>>> handlers = new HashMap<>();
+
+    public EndpointDataReferenceStoreReceiver(EndpointDataReferenceStore dataReferenceStore, PolicyArchive policyArchive, ContractAgreementService contractAgreementService, TransactionContext transactionContext, Monitor monitor) {
+        this.dataReferenceStore = dataReferenceStore;
+        this.policyArchive = policyArchive;
+        this.contractAgreementService = contractAgreementService;
+        this.transactionContext = transactionContext;
+        this.monitor = monitor;
+        registerHandler(TransferProcessStarted.class, this::handleTransferStarted);
+        registerHandler(TransferProcessTerminated.class, this::handleTransferTerminated);
+        registerHandler(TransferProcessCompleted.class, this::handleTransferCompleted);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <E extends Event> void on(EventEnvelope<E> event) {
+        if (event.getPayload() instanceof TransferProcessEvent transferProcessEvent && transferProcessEvent.getType().equals(TransferProcess.Type.CONSUMER.name())) {
+            var handler = (Function<E, Result<Void>>) handlers.get(transferProcessEvent.getClass());
+            if (handler != null) {
```

Comment:
nit: maybe logging a warning would be in order if no handler is found?

Reply:
Probably in this case it's not needed, if there is no handler is fine, we just skip that  `TransferProcessEvent`